### DISCUSSION TITLE
map.fitBounds(bounds) differs from map.getBounds() output.

### DISCUSSION THREAD

**Discussion by j** [maplibre/maplibre-gl-js#1779] [Q&A]
We're trying to save state of a map via user input and have come across an issue where when we save `map.getBounds` and re-create it, `map.getBounds` changes:

```typescript
    const map = new mapLibre.Map({
        container,
        bounds,
        style: getMapStyleSpecification(container, lineWidthScale),
        maxZoom: 14,
        minZoom: 2,
        preserveDrawingBuffer: true,
        attributionControl: false
    });

    map.fitBounds(bounds); // map.getBounds() does not match `bounds` given from above even though it came from previous `map.getBounds()` call
```

**Comment by HarelM** [comment]
Fit bounds can behave differently depending on the screen resolution (or more correctly the screen aspect ratio).
Get bounds get the corners of the screen.
So fit bounds and get bounds are not symmetrical operations.

**Reply by Emocody666** [reply]
How  long have you been stealing people's accounts

**Reply by HarelM** [reply]
@Emocody666 Is this message directed at me? I have no clue what do you mean by it...

**Reply by wipfli** [reply]
I don't think so, @HarelM...

@Emocody666 please clarify your message or remove it if it was a typo.

**Comment by j** [comment]
@HarelM yes, I understand and figured that. But how could I enforce exact bounds when aspect ratio does not change?

**Comment by HarelM** [comment]
How do you decide which bounds to set? I guess you can use some projection from screen pixels to lat lng...

**Comment by HarelM** [comment]
I'm not sure this is an issue though. Feels more like a discussion.
Can I convert it to a discussion?

**Comment by j** [comment]
I suppose it can be a discussion depending on how you look at it.

I feel that setting a map's bounds, getting those bounds, saving them in a DB and initializing that same map with those bounds that creates different bounds in the same exact container with set aspect ratio can potentially be an issue.  (Sorry for the run on sentence).  

We attempted react state and kept getting re-renders because props didn't match but I couldn't spend too much time on it.

**Comment by j** [comment]
@HarelM To me, this appears to be a bug/issue:

```typescript
const map1 = new Map({
    container,
    bounds,
    style
});

const map2 = new Map({
    container,
    bounds: map1.getBounds(),
    style
});

// map1.getBounds() != map2.getBounds().
```

Both maps have different bounds.  I confirmed that the container width/height & the map transformer width/height match and that it's not some weird issue from resizing.

**Reply by HarelM** [reply]
I would suggest to link a jsbin/stackblitz to show this issue. The above code has a lot of context missing.

**Comment by petr-hajek** [comment]
I came accross the same issue - calling fitBounds after getting bounds is zooming out - when the map has non-zero bearing or pitch. Repro: https://codesandbox.io/p/sandbox/fitbounds-lppksv
 I also found a workaround:
```
        const prevBearing = map.getBearing();
        const prevPitch = map.getPitch();
        map.jumpTo({ bearing: 0, pitch: 0 });
        const bounds = map.getBounds();
        const options = map.cameraForBounds(bounds);
// call this in another map, or when you want to restore a map to these bounds
// note: you also need to call this here or at least reverse the pitch and bearing changes above
        map.jumpTo({
          center: options.center,
          zoom: options.zoom,
          bearing: prevBearing,
          pitch: prevPitch,
        });
```
I'm not sure if this could be done differently, like reworking either the fitBounds or getBounds function
